rest-api-scripts/metric-reporter/workflow_metric_reporter.py

```diff
@@ -0,0 +1,132 @@
+            print("CREATED " + "PROJECT_" + project_id + "_LONGEST_LATENCY_BY_WORKFLOW_" + datetime.now().strftime(
+                "%Y-%m-%d at %H:%M:%S") + ".csv")
+            print("CREATING " + "PROJECT_" + project_id + "_STATUS_COUNT_" + datetime.now().strftime(
+                "%Y-%m-%d at %H:%M:%S") + ".csv")
+            get_status_count(total_reporter_dataframe, project_id)
+            print("CREATED " + "PROJECT_" + project_id + "_STATUS_COUNT_" + datetime.now().strftime(
+                "%Y-%m-%d at %H:%M:%S") + ".csv")
+            print("CREATING " + "PROJECT_" + project_id + "_OLDEST_EXECUTIONS" + datetime.now().strftime(
+                "%Y-%m-%d at %H:%M:%S") + ".csv")
+            get_oldest_executions(total_reporter_dataframe, project_id)
+            print("CREATED " + "PROJECT_" + project_id + "_OLDEST_EXECUTIONS" + datetime.now().strftime(
+                "%Y-%m-%d at %H:%M:%S") + ".csv")
+
+def get_longest_latency(total_reporter_dataframe,project_id):
+    workflow_dataframe = total_reporter_dataframe.groupby(["analysisFlowId","workflow_name"])["execution_time"].mean().sort_values(ascending=False)
+    export_name = "PROJECT_" + project_id + "_LONGEST_LATENCY_BY_WORKFLOW_" + datetime.now().strftime(
+        "%Y-%m-%d at %H:%M:%S") + ".csv"
+    workflow_dataframe.to_csv(export_name)
+
+def get_status_count(total_reporter_dataframe, project_id):
+    workflow_dataframe = total_reporter_dataframe.groupby(["analysisFlowId", "workflow_name","status"])["status"].count()
+    export_name = "PROJECT_" + project_id + "_STATUS_COUNT_" + datetime.now().strftime(
+        "%Y-%m-%d at %H:%M:%S") + ".csv"
+    workflow_dataframe.to_csv(export_name)
+
+def get_oldest_executions(total_reporter_dataframe, project_id):
+    workflow_dataframe = total_reporter_dataframe.sort_values(by=["endTime", 'analysisFlowId', "workflow_name"], ascending=[True, False, False])
+    export_name = "PROJECT_" + project_id + "_OLDEST_EXECUTIONS_" + datetime.now().strftime(
+        "%Y-%m-%d at %H:%M:%S") + ".csv"
+    workflow_dataframe.to_csv(export_name)
+
+def get_execution_information(fire_url, token, workflow_id):
+    get_execution_information_url = fire_url + "/api/v1/workflow-executions/workflows/" + workflow_id
+    api_call_headers = {'token': token}
+    status_code = requests.get(get_execution_information_url, headers=api_call_headers).status_code
+    if(status_code==200):
+        get_execution_information_text = requests.get(get_execution_information_url, headers=api_call_headers).text
+        get_execution_information_dict = json.loads(get_execution_information_text)
+        executionDataframe = pd.DataFrame()
+        for execution in get_execution_information_dict:
+            execution_id = str(execution["id"])
+            analysisFlowId = str(execution["analysisFlowId"])
+            workflow_name = execution["name"]
+            execution_time = execution["endTime"] - execution["startTime"]
```

Comment:
Assuming time is measured on the 24 hour format ie hours in a day vary from 00:00 AM to 24:00 PM

What happens if `startTime` is 23:00 PM the `endTime` is  01:00 AM. Then the logic will get `execution_time` of `01`-`23` = -22 most likely with the above logic?

Reply:
Ideally, you should be doing a datediff rather than a plain subtraction.

Reply:
Execution_time is in standard epoch time so the end time would always be greater than the start time unless the execution is running or if it failed, in which case the end time would be equal to the start time and execution_time would be 0.

Reply:
If this value should be of type datetime rather than of type int then I can do a datedif, do let me know.